src/walberla_bridge/src/lattice_boltzmann/InterpolateAndShiftAtBoundary.hpp

```diff
@@ -111,24 +119,30 @@ class InterpolateAndShiftAtBoundary {
     auto const prefactor =
         ((slab_dir == m_slab_max) ? FloatType{-1} : FloatType{1});
     auto const offset = get_pos_offset() * prefactor;
+    auto const folded_offset = python_modulo(offset, length);
+    auto const weight1 = 1 - std::fmod(folded_offset, FloatType{1});
+    auto const weight2 = std::fmod(folded_offset, FloatType{1});
     for (auto const &&cell : ci) {
       Cell source1 = cell;
       Cell source2 = cell;
-      source1[dir] = cell_idx_c(std::floor(
-                         static_cast<FloatType>(source1[dir]) + offset)) %
-                     dim;
-      source1[dir] = cell_idx_c(static_cast<FloatType>(source1[dir]) + length);
-      source1[dir] = cell_idx_c(source1[dir] % dim);
-
+      int target_idx = cell[dir];
+      auto ghost_fold = [dim](FloatType x) {
+        while (x > dim) {
```

Comment:
this also looks a lot like a modulo

Reply:
It isn't, actually. A modulo would be x>=n or x<n. Here, it's x>n or x<-1. This is, ebcause Walberla uses -1 and n to address the ghost layers.